src/main/cpp/mechanisms/ejectMechanism/generated/ejectMechanism.h

```diff
@@ -12,52 +13,44 @@
 // DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
 // OR OTHER DEALINGS IN THE SOFTWARE.
 //====================================================================================================================================================
+// This file was automatically generated by the Team 302 code generator version 1.1.0.0
+// Generated on Friday, January 5, 2024 7:16:48 AM
 
 #pragma once
 
-// C++ Includes
-#include <map>
-#include <memory>
 #include <string>
+#include <memory>
 
-// FRC includes
+// FRC Includes
+#include <networktables/NetworkTable.h>
 
-// Team 302 includes
+#include "mechanisms/base/BaseMech.h"
+#include "configs/RobotConfigMgr.h"
 
-// Third Party Includes
+#include "hw/DragonSolenoid.h"
 
-class CanSensorUsage
+class ejectMechanism : public BaseMech
 {
-
 public:
-	/// @enum CANSENSOR_USAGE
-	/// @brief Defines CanCoder usages.  This should be modified for each robot.
-	enum CANSENSOR_USAGE
-	{
-		UNKNOWN_CANSENSOR_USAGE = -1,
-
-		LEFT_FRONT_SWERVE_ANGLE,
-		RIGHT_FRONT_SWERVE_ANGLE,
-		LEFT_BACK_SWERVE_ANGLE,
-		RIGHT_BACK_SWERVE_ANGLE,
-		ARM_ANGLE,
-
-		PIGEON_ROBOT_CENTER,
-
-		EXAMPLE_CANCODER,
+	ejectMechanism() = delete;
+	~ejectMechanism() = default;
+	ejectMechanism ( MechanismTypes::MECHANISM_TYPE type, std::string networkTableName );
 
-		MAX_CANSENSOR_USAGES
-	};
+	virtual void Initialize ( RobotConfigMgr::RobotIdentifier robotFullName ) = 0;
+	void Cyclic();
 
-	static CanSensorUsage *GetInstance();
+	DragonSolenoid* ejectSolenoid;
+//state* state_1;
 
-	CANSENSOR_USAGE GetUsage(std::string usageString);
-	std::string GetUsage(CANSENSOR_USAGE usate);
+protected:
+	std::string m_ntName;
+	std::string m_tuningIsEnabledStr;
```

Comment:
Dave - This shouldn't be in the mechanism;  States are where the tuning needs to be.   Joe